Add RLNv2 to TheWakuNetwork (#82)
vacp2p/rfc-index: waku/standards/core/64/network.md

accessible even to resource-restricted devices.
We'll refer to this opinionated deployment simply as
_the public Waku Network_, _the Waku Network_ or, if the context is clear, _the network_

> in the rest of this document. All The Waku Network configuration parameters are listed [here](https://github.com/waku-org/nwaku/blob/master/waku/factory/networks_config.nim#L31).

Reviewer: ```suggestion
in the rest of this document.
All The Waku Network configuration parameters are listed [here](https://github.com/waku-org/nwaku/blob/8bfad3ab453f96ac545c7cb0af06d0c0f34d1356/waku/factory/networks_config.nim#L31).
```

Semantic breaks and permalinks for specs.

Author: done: fd45bb7495c9123dd2f00becf30643db5660e6e3

## RLN rate-limiting

> The [17/WAKU2-RLN-RELAY](../17/rln-relay.md) protocol uses [32/RLN-V1](../../../../vac/32/rln-v1.md), extended with [RLN-V2](https://github.com/vacp2p/rfc-index/blob/main/vac/raw/rln-v2.md) proofs

Reviewer: Shouldn't this now just be relay "...protocol uses RLN-V2" proofs? In other words, is RLN-V1 still relevant to be mentioned?

Author: indeed, rln-v1 is not relevant anymore: fd45bb7495c9123dd2f00becf30643db5660e6e3

> to ensure that a pre-agreed rate limit is not exceeded by any publisher. The addition of RLNv2 allows to rate-limit with more granularity, `x` amount of messages every `y` seconds.

Reviewer: ```suggestion
to ensure that a pre-agreed rate limit is not exceeded by any publisher.
The addition of RLN-V2 allows rate-limiting with more granularity, `x` amount of messages every `y` seconds.
```

As for `RLN-V2` vs `RLNv2` vs `RLNV2` (here and elsewhere) I have no particular preference as long as we use it consistently

Author: fix in fd45bb7495c9123dd2f00becf30643db5660e6e3 using RLN-V2

* `rlnRelayUserMessageLimit=100`: Amount of messages that a membership is allowed to publish per epoch. Configurable between `0` and `MAX_MESSAGE_LIMIT`.
* `rlnEpochSizeSec=600`: Size of the epoch in seconds.
* `rlnRelayChainId=11155111`: Network in which the RLN contract is deployed, aka Sepolia.
* `rlnRelayEthContractAddress=0xCB33Aa5B38d79E3D9Fa8B10afF38AA201399a7e3`: Network address where RLN memberships are stored.
* `staked_fund=0`: In other words, the Waku Network does not use RLN staking. Registering a membership just requires to pay gas.
* `MAX_MESSAGE_LIMIT=100`: Maximum amount of messages allowed per epoch for any membership. Enforced in the contract.
* `max_epoch_gap=20`: Maximum allowed gap in seconds into the past or future compared to the validator's clock.

Nodes MUST _reject_ messages not respecting any of these parameters.
Nodes SHOULD use Network Time Protocol (NTP) to synchronize their own clocks,
thereby ensuring valid timestamps for proof generation and validation.
Publishers to the Waku Network SHOULD register an RLN membership.

### RLN Proofs

Each RLN member MUST generate and attach an RLN proof to every published message

> as described in [17/WAKU2-RLN-RELAY](../17/rln-relay.md/#publishing) and [RLNV2](https://github.com/vacp2p/rfc-index/blob/main/vac/raw/rln-v2.md).

Reviewer: ```suggestion
as described in [17/WAKU2-RLN-RELAY](../17/rln-relay.md/#publishing) and [RLN-V2](https://github.com/vacp2p/rfc-index/blob/main/vac/raw/rln-v2.md).
```

Author: fixed: fd45bb7495c9123dd2f00becf30643db5660e6e3

If a message contains an RLN proof
and the zero-knowledge proof is invalid

> according to the verification process described in [32/RLN-V1](../../../../vac/32/rln-v1.md)/[RLN-V2](https://github.com/vacp2p/rfc-index/blob/main/vac/raw/rln-v2.md?plain=1),

Reviewer: Same question as before - shouldn't we rather drop all references to RLN-V1?

Author: dropped the reference: fd45bb7495c9123dd2f00becf30643db5660e6e3
